uportal-war/src/main/java/org/springframework/context/support/ExtendedPropertySourcesPlaceholderConfigurer.java

```diff
@@ -37,6 +42,30 @@
     public static final String EXTENDED_PROPERTIES_SOURCE = "extendedPropertiesSource";
 
     private PropertyResolver propertyResolver;
+    private final Logger logger = LoggerFactory.getLogger(getClass());
+
+    /**
+     * uPortal defines some properties files in its primaryPropertyPlaceholderConfigurer
+     * bean that are considered (and documented) optional.  The parent class
+     * (PropertySourcesPlaceholderConfigurer) will operate properly without them, but
+     * puts a significant number of WARN messages into the log.  These are noisy, and
+     * could lead a new adopter to think that something's wrong.  This method removes
+     * absent properties files from the collection.
+     */
+    @Override
+    public void setLocations(Resource[] locations) {
+        final List<Resource> list = new ArrayList<>();
+        for (Resource r : locations) {
+            if (r.exists()) {
+                list.add(r);
+            } else {
+                // In our case this event is worth a DEBUG note.
+                logger.debug("The following Resource was not present (it may be "
+                        + "optional, or it's absence may lead to issues):  ", r);
```

Comment:
I suggest checking the 'ignoreResourceNotFound' property which we set to true in applicationContext.xml and if that property is set to false (the default), call the superclass method.  This code effectively removes the missing resource and Spring will not fail as it normally would in PropertiesLoaderSupport.loadProperties().  Only ignore the resource and issue a debug message if that property is set to true.  That maintains the spirit of the Spring contract.